### PR TITLE
A new helm-chart - veritable-ui

### PR DESCRIPTION
**Pull request by n3op2**
# What

a helm chart for veritable-ui including a couple of secrets:
- companysHouseApi
- cookieSessionKeys
along with other basics/defaults for full details a pod describe is below

JIRA -> https://digicatapult.atlassian.net/browse/VR-89

```eature/vr-69) $ kubectl describe pod test-veritable-ui-d9774dfdc-28s8v
Name:             test-veritable-ui-d9774dfdc-28s8v
Namespace:        default
Priority:         0
Service Account:  test-veritable-ui
Node:             veritable-control-plane/172.18.0.2
Start Time:       Thu, 23 May 2024 09:00:26 +0100
Labels:           app.kubernetes.io/component=veritable-ui
                  app.kubernetes.io/instance=test
                  app.kubernetes.io/managed-by=Helm
                  app.kubernetes.io/name=veritable-ui
                  app.kubernetes.io/version=v0.3.7
                  helm.sh/chart=veritable-ui-1.0.0
                  pod-template-hash=d9774dfdc
Annotations:      <none>
Status:           Running
IP:               10.244.0.5
IPs:
  IP:           10.244.0.5
Controlled By:  ReplicaSet/test-veritable-ui-d9774dfdc
Init Containers:
  database-migrate:
    Container ID:  containerd://93f66ef29407bbbec2ab38a6ab164df4f0ba64bfc4357e38992567ef9a6185c7
    Image:         docker.io/digicatapult/veritable-ui:v0.3.7
    Image ID:      docker.io/digicatapult/veritable-ui@sha256:407f48f6ee88c639ec53b0fa2b6558b7f3dc28570920333c57e466e757360df7
    Port:          <none>
    Host Port:     <none>
    Command:
      npx
      knex
    Args:
      migrate:latest
      --knexfile
      knexfile.js
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Thu, 23 May 2024 09:03:02 +0100
      Finished:     Thu, 23 May 2024 09:03:02 +0100
    Ready:          True
    Restart Count:  4
    Environment:
      NODE_ENV:     production
      DB_HOST:      test-postgresql
      DB_PORT:      5432
      DB_NAME:      veritable-ui
      DB_USERNAME:  veritable-ui
      DB_PASSWORD:  <set to the key 'password' in secret 'test-postgresql'>  Optional: false
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-2snhl (ro)
Containers:
  veritable-ui:
    Container ID:   containerd://1ff43a439d48c3844f3b64aa4e7837ef04b5013a965783edfe9c530a87fd23da
    Image:          docker.io/digicatapult/veritable-ui:v0.3.7
    Image ID:       docker.io/digicatapult/veritable-ui@sha256:407f48f6ee88c639ec53b0fa2b6558b7f3dc28570920333c57e466e757360df7
    Port:           3000/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Thu, 23 May 2024 09:03:03 +0100
    Ready:          True
    Restart Count:  0
    Environment:
      COOKIE_SESSION_KEYS:      [secret]
      PUBLIC_URL:               http://localhost:3000
      API_SWAGGER_BG_COLOR:     #fafafa
      API_SWAGGER_TITLE:        Veritable
      API_SWAGGER_HEADING:      Veritable
      IDP_CLIENT_ID:            veritable-ui
      IDP_CLIENT_ID:            veritable-ui
      IDP_PUBLIC_URL_PREFIX:    http://localhost:3080/realms/veritable/protocol/openid-connect
      IDP_INTERNAL_URL_PREFIX:  http://localhost:3080/realms/veritable/protocol/openid-connect
      IDP_AUTH_PATH:            /auth
      IDP_TOKEN_PATH:           /token
      IDP_JWKS_PATH:            /certs
      COMPANY_HOUSE_API_URL:    https://api.company-information.service.gov.uk
      COMPANY_PROFILE_API_KEY:  companysHouseApiKey
      PORT:                     3000
      LOG_LEVEL:                debug
      DB_HOST:                  test-postgresql
      DB_PORT:                  5432
      DB_NAME:                  veritable-ui
      DB_USERNAME:              veritable-ui
      DB_PASSWORD:              <set to the key 'password' in secret 'test-postgresql'>  Optional: false
      EXTERNAL_ORIGIN:          
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-2snhl (ro)
Conditions:
  Type                        Status
  PodReadyToStartContainers   True 
  Initialized                 True 
  Ready                       True 
  ContainersReady             True 
  PodScheduled                True 
Volumes:
  kube-api-access-2snhl:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason     Age                    From               Message
  ----     ------     ----                   ----               -------
  Normal   Scheduled  7m21s                  default-scheduler  Successfully assigned default/test-veritable-ui-d9774dfdc-28s8v to veritable-control-plane
  Normal   Pulling    7m21s                  kubelet            Pulling image "docker.io/digicatapult/veritable-ui:v0.3.7"
  Normal   Pulled     6m17s                  kubelet            Successfully pulled image "docker.io/digicatapult/veritable-ui:v0.3.7" in 1m4.342s (1m4.343s including waiting)
  Warning  BackOff    5m (x8 over 6m15s)     kubelet            Back-off restarting failed container database-migrate in pod test-veritable-ui-d9774dfdc-28s8v_default(1072df3a-6654-4520-acc8-62d82828d5b0)
  Normal   Created    4m45s (x5 over 6m17s)  kubelet            Created container database-migrate
  Normal   Started    4m45s (x5 over 6m17s)  kubelet            Started container database-migrate
  Normal   Pulled     4m45s (x4 over 6m16s)  kubelet            Container image "docker.io/digicatapult/veritable-ui:v0.3.7" already present on machine
  Normal   Pulled     4m44
```

### CODE REVIEW

File: charts/veritable-ui/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+{{- define "veritable-ui.databaseHost" -}}
+{{- ternary (include "veritable-ui.postgresql.fullname" .) .Values.externalDatabase.host .Values.postgresql.enabled | quote -}}
+{{- end -}}
+
+{{/*
+Return the company profile API key secret name
+*/}}
+{{- define "veritable-ui.companyHouseSecret" -}}
+{{- ternary (include "veritable-ui.companyHouse.fullname" .) .Values.companyHouse.secret .Values.postgresql.enabled | quote -}}
+{{- end -}}
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/templates/_helpers.tpl:45]: Don't understand what this is going to achieve.

Comment (n3op2) [reply charts/veritable-ui/templates/_helpers.tpl:45]: put in `companyHouseSecret` variable for later usage, did follow postgres "existing secret config"

File: charts/veritable-ui/values.yaml [hunk 2]
```diff
@@ -0,0 +1,701 @@
...
+##
+## @param companyHouse.url an external API from which company details can be retrieved
+## @param companyHouse.apiKey an API key
+## @param companyHouse.secret name of existing secret to use for company house API key
+
+companyHouse:
+  enabled: true
+  url: "https://api.company-information.service.gov.uk"
+  apiKey: ""
+  secret: ""
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/values.yaml:583]: Move this to just under the global parameters section and create a `## @section veritable-cloudagent configuration` replacing what is on line 63

File: charts/veritable-ui/values.yaml [hunk 3]
```diff
@@ -0,0 +1,701 @@
...
+  port: ""
+
+## @section SQNC-Identity-Service parameters
+## @param externalSqncIdentity.host External SQNC-Identity-Service hostname
+##
+externalSqncIdentity:
+  host: ""
+  ## @param externalSqncIdentity.port External SQNC-Identity-Service port
+  ##
+  port: ""
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/values.yaml:682]:
```suggestion
```

File: charts/veritable-ui/values.yaml [hunk 4]
```diff
@@ -0,0 +1,701 @@
...
+## @param auth.idp.JWKSPath IDP JWKS path
+auth:
+  enabled: true
+  idp: 
+    clientId: "veritable-ui"
+    publicURLPrefix: "http://localhost:3080/realms/veritable/protocol/openid-connect"
+    internalURLPrefix: "http://localhost:3080/realms/veritable/protocol/openid-connect"
+    authPath: "/auth"
+    tokenPath: "/token"
+    JWKSPath: "/certs"
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/values.yaml:701]: Move all of this to the new Veritable-ui config section

File: charts/veritable-ui/values.yaml [hunk 5]
```diff
@@ -591,6 +587,7 @@ serviceAccount:
 ## @param postgresql.architecture PostgreSQL architecture (`standalone` or `replication`)
 ##
 postgresql:
+  fullName: "veritable-ui"
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/values.yaml:590]:
```suggestion
```

File: charts/veritable-ui/values.yaml (hunk 4 above)
Comment (n3op2) [reply charts/veritable-ui/values.yaml]: this has been deleted

File: charts/veritable-ui/values.yaml (hunk 2 above)
Comment (n3op2) [reply charts/veritable-ui/values.yaml]: this should have been moved

File: charts/veritable-ui/templates/_helpers.tpl (hunk 1 above)
Comment (n3op2) [reply charts/veritable-ui/templates/_helpers.tpl]: this is also should have been updated

**Review by mattdean-digicatapult — DISMISSED** [review]
I'd also like to wait for Dave before merging this, but we're getting really close

File: charts/veritable-ui/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  - condition: postgresql.enabled
+    name: postgresql
+    repository: https://charts.bitnami.com/bitnami
+    version: 15.x.x
+  - name: common
+    repository: https://charts.bitnami.com/bitnami
+    tags:
+      - bitnami-common
+    version: 2.x.x
+description: The veritable user interface for managing certificates
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/Chart.yaml:16]:
```suggestion
description: The veritable user interface for managing and creating connections
```

File: charts/veritable-ui/Chart.yaml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+home: https://github.com/digicatapult/veritable-ui
+icon: https://raw.githubusercontent.com/digicatapult/sqnc-documentation/main/assets/icon.png
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/Chart.yaml:18]: Check to see if we have an icon for this in `veritable-documentation` if not, remove this line.

File: charts/veritable-ui/templates/NOTES.txt [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+  kubectl get pods --namespace {{ include "common.names.namespace" . | quote }} -l app.kubernetes.io/instance={{ .Release.Name }}
+
+Access the pod you want to debug by executing
+
+  kubectl exec --namespace {{ include "common.names.namespace" . | quote }} -ti <NAME OF THE POD> -- bash
+
+In order to replicate the container startup scripts execute this command:
+
+    node ./app/index.js
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/templates/NOTES.txt:23]:
```suggestion
node build/index.js
```

File: charts/veritable-ui/templates/deployment.yaml [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "veritable-ui.databaseSecretName" . }}
+                  key: {{ include "veritable-ui.databaseSecretPasswordKey" . }}
+            - name: EXTERNAL_ORIGIN
+              value: {{ .Values.externalOrigin }}
+            {{ if .Values.externalPathPrefix }}
+            - name: EXTERNAL_PATH_PREFIX
+              value: {{ .Values.externalPathPrefix }}
+            {{ end }}
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/templates/deployment.yaml:209]:
```suggestion
```

File: charts/veritable-ui/values.yaml [hunk 10]
```diff
@@ -0,0 +1,643 @@
...
+  command:
+    - sleep
+  ## @param diagnosticMode.args Args to override all containers in the deployment
+  ##
+  args:
+    - infinity
+
+## @section veritable cloudagent config parameters
+## TODO update name to companys
+## TODO bash completion make sure kubectl (bash complete@2)
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/values.yaml:65]:
```suggestion
```

File: charts/veritable-ui/values.yaml [hunk 11]
```diff
@@ -0,0 +1,643 @@
...
+  ## @param diagnosticMode.command Command to override all containers in the deployment
+  ##
+  command:
+    - sleep
+  ## @param diagnosticMode.args Args to override all containers in the deployment
+  ##
+  args:
+    - infinity
+
+## @section veritable cloudagent config parameters
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/values.yaml:63]:
```suggestion
## @section veritable ui config parameters
```

File: charts/veritable-ui/values.yaml [hunk 12]
```diff
@@ -0,0 +1,643 @@
...
+apiSwaggerBgColor: "#fafafa"
+apiSwaggerTitle: "Veritable"
+apiSwaggerHeading: "Veritable"
+
+idpClientId: "veritable-ui"
+idpPublicURLPrefix: "http://localhost:3080/realms/veritable/protocol/openid-connect"
+idpInternalURLPrefix: "http://localhost:3080/realms/veritable/protocol/openid-connect"
+idpAuthPath: "/auth"
+idpTokenPath: "/token"
+idpJWKSPath: "/certs"
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/values.yaml:92]: For all the values you have added, you need to add the [bitnami readme generator](https://github.com/bitnami/readme-generator-for-helm?tab=readme-ov-file#valuesyaml-metadata) to this file and ensure that they make sense.

File: charts/veritable-ui/values.yaml [hunk 13]
```diff
@@ -0,0 +1,643 @@
...
+## @param livenessProbe.path Path for to check for livenessProbe
+## @param livenessProbe.initialDelaySeconds Initial delay seconds for livenessProbe
+## @param livenessProbe.periodSeconds Period seconds for livenessProbe
+## @param livenessProbe.timeoutSeconds Timeout seconds for livenessProbe
+## @param livenessProbe.failureThreshold Failure threshold for livenessProbe
+## @param livenessProbe.successThreshold Success threshold for livenessProbe
+##
+livenessProbe:
+  enabled: false
+  path: /health
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/values.yaml:144]:
```suggestion
livenessProbe:
  enabled: enabled
  path: /
```

For now we will just do a probe off of the root.

File: charts/veritable-ui/values.yaml [hunk 14]
```diff
@@ -0,0 +1,643 @@
...
+## @param readinessProbe.path Path for to check for readinessProbe
+## @param readinessProbe.initialDelaySeconds Initial delay seconds for readinessProbe
+## @param readinessProbe.periodSeconds Period seconds for readinessProbe
+## @param readinessProbe.timeoutSeconds Timeout seconds for readinessProbe
+## @param readinessProbe.failureThreshold Failure threshold for readinessProbe
+## @param readinessProbe.successThreshold Success threshold for readinessProbe
+##
+readinessProbe:
+  enabled: false
+  path: /health
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/values.yaml:160]: Again, enable and `/`

File: charts/veritable-ui/values.yaml [hunk 15]
```diff
@@ -0,0 +1,643 @@
...
+## @param startupProbe.path Path for to check for startupProbe
+## @param startupProbe.initialDelaySeconds Initial delay seconds for startupProbe
+## @param startupProbe.periodSeconds Period seconds for startupProbe
+## @param startupProbe.timeoutSeconds Timeout seconds for startupProbe
+## @param startupProbe.failureThreshold Failure threshold for startupProbe
+## @param startupProbe.successThreshold Success threshold for startupProbe
+##
+startupProbe:
+  enabled: false
+  path: /health
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/values.yaml:176]: Again, enable and `/`

File: charts/veritable-ui/values.yaml [hunk 16]
```diff
@@ -0,0 +1,643 @@
...
+  ##
+  ingressClassName: ""
+  ## @param ingress.paths [array] Default paths for the ingress record
+  ## e.g:
+  ## paths:
+  ##   - path: /
+  ##     pathType: Prefix
+  paths:
+    - path: /connection
+      pathType: Prefix
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/values.yaml:450]:
```suggestion
```
If we are using `/` as a path, then this is redundant. See [here](https://kubernetes.io/docs/concepts/services-networking/ingress/#path-types)

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]
This is good.  I have left review comments and suggestions where I can in the codebase.

The only other task that needs to be done is the creation of a `README.md` in the `charts/veritable-ui` folder.  You should copy over the `veritable-cloudagent` one and edit appropriately.  Then you can use the `bitnami-helm-readme-generator` (NPM package available) to generate the table of parameters and their defaults for the `README.md`

Once that is done, please update the `README.md` in the root of the repository with a new entry for this helm chart.

File: charts/veritable-ui/values.yaml (hunk 12 above)
Comment (n3op2) [reply charts/veritable-ui/values.yaml:90]: on it! thanks

File: charts/veritable-ui/Chart.yaml (hunk 7 above)
Comment (n3op2) [reply charts/veritable-ui/Chart.yaml:18]: ok, will confirm on this shortly

File: charts/veritable-ui/values.yaml (hunk 14 above)
Comment (n3op2) [reply charts/veritable-ui/values.yaml:158]: done

File: charts/veritable-ui/templates/deployment.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (n3op2) [reply charts/veritable-ui/templates/deployment.yaml]: this should have been removed

File: charts/veritable-ui/README.md [hunk 17]
```diff
@@ -0,0 +1,309 @@
+# veritable-ui
+
+The veritable-ui is a component of the Sequence (veritable) ledger-based system. The veritable-ui responsible for matching orders, and transactions as well as indexing blocks locally, it exposes API for this purpose. See [https://github.com/digicatapult/veritable-documentation](https://github.com/digicatapult/veritable-documentation) for details.
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/README.md:3]: Change the description

File: charts/veritable-ui/README.md [hunk 18]
```diff
@@ -0,0 +1,309 @@
...
+To install the chart with the release name `my-release`:
+
+```console
+$ helm repo add digicatapult https://digicatapult.github.io/helm-charts
+$ helm install my-release digicatapult/veritable-ui
+```
+
+The command deploys veritable-ui on the Kubernetes cluster in the default configuration. The [Parameters](#parameters) section lists the parameters that can be configured during installation.
+
+You will need to specify the `externalveritableIpfs` host and port as well as the `externalveritableIdentityService` host and port.
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/README.md:33]: Remove.

File: charts/veritable-ui/values.yaml [hunk 19]
```diff
@@ -0,0 +1,655 @@
...
+## @param idpInternalURLPrefix veritable-ui private/internal realm
+idpInternalURLPrefix: "http://localhost:3080/realms/veritable/protocol/openid-connect"
+## @param idpAuthPath veritable-ui IDP authentication path
+idpAuthPath: "/auth"
+## @param idpTokenPath veritable-ui IDP token path
+idpTokenPath: "/token"
+## @param idpJWKSPath veritable-ui IDP certs path
+idpJWKSPath: "/certs"
+## Digital Catapult veritable-ui image
+## ref: https://hub.docker.com/r/digicatapult/veritable-ui/tags/
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/values.yaml:109]:
```suggestion
## @section veritable-ui deployment parameters
## Digital Catapult veritable-ui image
## ref: https://hub.docker.com/r/digicatapult/veritable-ui/tags/
```

File: charts/veritable-ui/values.yaml [hunk 20]
```diff
@@ -0,0 +1,655 @@
...
+## @param livenessProbe.path Path for to check for livenessProbe
+## @param livenessProbe.initialDelaySeconds Initial delay seconds for livenessProbe
+## @param livenessProbe.periodSeconds Period seconds for livenessProbe
+## @param livenessProbe.timeoutSeconds Timeout seconds for livenessProbe
+## @param livenessProbe.failureThreshold Failure threshold for livenessProbe
+## @param livenessProbe.successThreshold Success threshold for livenessProbe
+##
+livenessProbe:
+  enabled: true
+  path: /
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/values.yaml:158]:
```suggestion
  path: /connections
```

File: charts/veritable-ui/values.yaml [hunk 21]
```diff
@@ -0,0 +1,655 @@
...
+## @param readinessProbe.path Path for to check for readinessProbe
+## @param readinessProbe.initialDelaySeconds Initial delay seconds for readinessProbe
+## @param readinessProbe.periodSeconds Period seconds for readinessProbe
+## @param readinessProbe.timeoutSeconds Timeout seconds for readinessProbe
+## @param readinessProbe.failureThreshold Failure threshold for readinessProbe
+## @param readinessProbe.successThreshold Success threshold for readinessProbe
+##
+readinessProbe:
+  enabled: true
+  path: /
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/values.yaml:174]:
```suggestion
  path: /connections
```

File: charts/veritable-ui/values.yaml [hunk 22]
```diff
@@ -0,0 +1,655 @@
...
+## @param startupProbe.path Path for to check for startupProbe
+## @param startupProbe.initialDelaySeconds Initial delay seconds for startupProbe
+## @param startupProbe.periodSeconds Period seconds for startupProbe
+## @param startupProbe.timeoutSeconds Timeout seconds for startupProbe
+## @param startupProbe.failureThreshold Failure threshold for startupProbe
+## @param startupProbe.successThreshold Success threshold for startupProbe
+##
+startupProbe:
+  enabled: true
+  path: /
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/values.yaml:190]:
```suggestion
  path: /connections
```

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]

File: charts/veritable-ui/README.md (hunk 17 above)
Comment (n3op2) [reply charts/veritable-ui/README.md:3]: Updated and should be reflected in the next push

File: charts/veritable-ui/README.md (hunk 18 above)
Comment (n3op2) [reply charts/veritable-ui/README.md:33]: should be gone.

File: charts/veritable-ui/Chart.yaml (hunk 7 above)
Comment (n3op2) [reply charts/veritable-ui/Chart.yaml]: it does not exists and I've removed the "icon" property all together

File: charts/veritable-ui/templates/deployment.yaml [hunk 23]
```diff
@@ -0,0 +1,261 @@
...
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.command "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.diagnosticMode.enabled }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.args "context" $) | nindent 12 }}
+          {{- else if .Values.args }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.args "context" $) | nindent 12 }}
+          {{- end }}
+          env:
+            - name: COOKIE_SESSION_KEYS
+              value: {{ .Values.cookieSessionKeys.secret | quote }}
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/templates/deployment.yaml:160]:
```suggestion
            - name: COOKIE_SESSION_KEYS
              valueFrom:
                secretKeyRef:
                  name: {{ include "veritable-ui.cookieSessionKeysSecretName" . }}
                  key: {{ include "veritable-ui.cookieSessionKeysSecretKey" . }}
```

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]

File: charts/veritable-ui/templates/deployment.yaml [hunk 24]
```diff
@@ -0,0 +1,261 @@
...
+            - name: IDP_AUTH_PATH
+              value: {{ .Values.idpAuthPath | quote }}
+            - name: IDP_TOKEN_PATH
+              value: {{ .Values.idpTokenPath | quote }}
+            - name: IDP_JWKS_PATH
+              value: {{ .Values.idpJWKSPath | quote }}
+            - name: COMPANY_HOUSE_API_URL
+              value: {{ .Values.companysHouseApiUrl | quote }}
+            - name: COMPANY_PROFILE_API_KEY
+              value: {{ .Values.companysHouseApiKey.secret | quote }}
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/templates/deployment.yaml:189]:
```suggestion
            - name: COMPANY_PROFILE_API_KEY
              valueFrom:
	                secretKeyRef:
	                  name: {{ include "veritable-ui.companyHouseApiKeySecretName" . }}
	                  key: {{ include "veritable-ui.companyHouseApiKeySecretKey" . }}
```

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]

**Review by dblane-digicatapult — APPROVED** [review]

**Review by dblane-digicatapult — APPROVED** [review]

**Pull request merged**